### DISCUSSION TITLE
5.2 Changes not saved when I reopen codespace

### DISCUSSION THREAD

**Discussion by tracyblanchard** [ua-gist604b-f24/syllabus#18] [General]
![Screenshot 2024-11-14 185444](https://github.com/user-attachments/assets/122bf67f-e140-4f73-9bb2-d16a41f6c010)

When reopening codespace my changes aren't saved. I completed the tutorial from lecture and everything went smooth, but when returning to complete the exercises my data wasn't saved. My connection is still there and my database is still there but nothing in it, there is an error message in the bottom right when I click on the nyc database. I will probably just redo the tutorial but is there a way to save changes like this on codespace? this is not the only time I have had to redo my steps on codespace. thanks!

**Comment by kev-bone** [comment]
This happened to me yesterday as well. I right clicked on nyc-postgis and then Delete Connection. Then I hit the + sign and typed in the hostname, user, password, port and all that to connect again. And then I think it said nyc already exists, and then all the data was back. I can't claim this is the proper way to do it, but it worked that time at least.

**Reply by tracyblanchard** [reply]
Thanks for the support! I had to redo the first three steps again of the assignment but then my tables were still in there once I reconnected.

**Comment by travisz09** [comment]
Make sure that you are committing all your changes, then pushing back to the assignment branch. That way, no matter what environment you are in, even if you start a brand new codespace, all of your changes will be saved on the branch. You may have to set up the codespace environment again, but any data or edits that you made will be saved.
